File: src/main/java/model/rules/firsttoplay/BiggestDouble.java
package model.rules.firsttoplay;

import model.players.Player;
import java.util.List;

public class BiggestDouble implements FirstToPlay {

    @Override
    public int execute(List<Player> players) {
        boolean found = false;
        for (int i = 6; i >= 0 && !found; i--) {
            for (int j = 0; !found && j < players.size(); j++) {
                Player player = players.get(j);
                int n = i;
                if (player.getTiles().stream().anyMatch(p -> p.isDouble() && p.getFirstNumber() == n)) {
                    return j;
                }
            }
        }
        return 0;
    }
}

File: docs/scripts/model/rules/firsttoplay/BiggestDouble.js
export default class BiggestDouble {

    execute(players) {
        let found = false;
        for (let i = 6; i >= 0 && !found; i--) {
            for (let j = 0; !found && j < players.size(); j++) {
                let player = players[j];
                let n = i;
                if (player.getTiles().some(p => p.isDouble() && p.getFirstNumber() === n)) {
                    return j;
                }
            }
        }
        return 0;
    }
    
}
